step-plans/step-plans-core/src/main/java/step/core/execution/ExecutionEngineRunner.java

```diff
@@ -211,11 +213,27 @@ private void exportExecution(ExecutionContext context) {
 		}
 	}
 
-	private void persistReportNode(ReportNode reportNode) {
-		executionContext.getReportNodeAccessor().save(reportNode);
-	}
-	
 	private void updateStatus(ExecutionStatus newStatus) {
 		executionLifecycleManager.updateStatus(newStatus);
 	}
+
+	private void saveFailureReportWithStatus(ReportNodeStatus status) {
+		ReportNode contextReport = executionContext.getReport();
+		contextReport.setStatus(status);
+		executionContext.getReportNodeAccessor().save(contextReport);
+		// there must be at least one child element to the (context) report.
+		ReportNode rootReport = new ReportNode();
+		rootReport.setExecutionID(contextReport.getExecutionID());
+		rootReport.setParentID(contextReport.getId());
+		rootReport.setStatus(status);
+		try {
+			// This needs to be done like this -- it will work at runtime, but
+			// would create a compile-time cyclic dependency between modules if the dependencies were setup "correctly" in pom.xml
+			AbstractArtefact failure = (AbstractArtefact) Class.forName("step.artefacts.Failure").newInstance();

```

Comment:
Why do we need a report node for this? ReportNode are meant to report execution of artifacts. In the cas of an import error we have not plan to execute and thus no report node. Aren't we misusing the ReportNodes to have a result for the execution?

Reply:
Because the report node is used to show the icon on the UI. Otherwise, you woult not have the < x > icon...